src/main/java/BinarySearchTree.java

```diff
@@ -1,133 +1,66 @@
-import java.util.Iterator;
+import java.util.NoSuchElementException;
 
-public interface BinarySearchTree<T extends Comparable<? super T>> extends Iterable<T> {
+/**
+ * A binary search tree is a binary tree with the property that all elements to the left are less than the element in
+ * the root node and all elements to the right are greater than or equal to the element in the root node.
+ *
+ * @param <T> Type of elements that are to be in the binary search tree.
+ */
+public interface BinarySearchTree<T extends Comparable<? super T>> extends BinaryTree<T>, Iterable<T> {
```

Comment:
`BinaryTree<T>` already extends `Iterable<T>`, there's no need to specify it twice here